### PR TITLE
Translation editor

### PR DESCRIPTION
**Pull request by jonnieZG**
- [x] The commit message make sense
- [x] The changes have been tested locally
- [ ] New features have been documented in the Wiki
- [ ] Implemented JSON to Translations.cpp converter script
- [ ] Implemented JSON to new multi-lingual Translations.cpp and Font.cpp
- [x] I'm willing to maintain this in the future (Totally Optional)

**What kind of change does this PR introduce?** 
- A set of user-friendly tools for editing translations and maintaining separate JSON files for each language, regarding #222.

**Translation Editor**
- An HTML5 application that can be executed in Chrome, Firefox, MSIE or Edge, for editing language strings
- Automatically checks for length constraints (shown in red if overflown)
- Automatically marks empty strings, or strings that have the same value as the referent language

*Usage*
1. First load a referent language (for example EN)
2. Then load your language that you want to edit
3. Pay attention to fields marked in yellow. It means that their values are either empty, or the same as in the original (referent) file, so you probably want to consider translating them.
4. When done, press "Save" button to save the translation_xx.json file.
- If that should not work (in some browsers), then click "View", and then copy/paste the text into a new file named as above.
5. The generated JSON files should be compiled into the target Translation.cpp file as a step in make file. by a Python script (that I will write soon).

*Adding a New Language*
1. Select a language that is similar to your new target language.
2. Copy that fille and name it translation_XX.json (XX being the code of your new target language)
3. Start the Translation Editor and translate the strings.
4. Press "Save" (or "View") to save your new file.

**Translations Parser**
I made this HTML5 tool to initially parse the original Translation.cpp and generate a separate JSON file for each language.

### CODE REVIEW

**Comment by jonnieZG** [comment]
I have just added a ```make_translation.py``` python pre-compile script that creates ```Translation.cpp``` from JSON files. This one is just for single-language version, until we make a multi-lingual one. You can call it from the ```make``` file.

**CAUTION: The current Translation.cpp file will be overwritten!**

```make_translation.py``` accepts two parameters: 
* first param is a directory containing ```translations_def.js``` and ```translation_*.json``` files;
* second parameter is a directory where the ```Translation.cpp``` will be created

File: Translation Editor/translation_lt.json [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+		"ScrollingSpeed": {
+			"text": "",
+			"text2": [
+				"Greitas apra-",
+				"šymo slinkimas"
+			],
+			"desc": "Greitis, kuriuo šis tekstas slenka"
+		}
+	},
+	"languageLocalName": "Lietuviškai"
```
Comment (jvitkauskas) [inline Translation Editor/translation_lt.json]: Should be `"languageLocalName": "Lietuvių"`

Comment (jonnieZG) [reply Translation Editor/translation_lt.json]: Thanks, @jvitkauskas! That was Google translate. ;) Fixed it. Probably there will be more errors like this one on a few other languages as well.

Comment (jvitkauskas) [reply Translation Editor/translation_lt.json]: Hi, you have missed a comma after "languageLocalName": "Lietuvių". JSON is not valid anymore.

Comment (jonnieZG) [reply Translation Editor/translation_lt.json]: Yup, you are right again. Fixed. Sorry for that.

**Comment by Ralim** [comment]
Hey @jonnieZG 
Sorry I've been totally silent on this.

Where is this up to? Should we move over to this now :) ?

**Comment by Ralim** [comment]
Going to Merge this in @jonnieZG and look into integrating some of its features

**Comment by discip** [comment]
@jonnieZG
Good evening Sir,
since you are the creator of the [TranslationEditor.html](https://github.com/Ralim/IronOS/blob/dev/Translations/TranslationEditor.html),

I would like to know if there is a way to change the `TranslationEditor` to generate this:

```json
    "SolderingMenu": {
      "text2": ["Soldering", "settings"],
      "desc": "Soldering settings"
    },
```

instead of this:
```json
		"SolderingMenu": {
			"text2": [
				"Soldering",
				"settings"
			],
			"desc": "Soldering settings"
		},
```
Hopefully you are doing well and have time at hand to help me with that. 😊

thanks in advance
kind regards

**Comment by discip** [comment]
@jonnieZG
Good evening Sir,
since you are the creator of the [TranslationEditor.html](https://github.com/Ralim/IronOS/blob/dev/Translations/TranslationEditor.html),

1. I would like to know if there is a way to change the **`TranslationEditor`** to generate this:

   ```json
    "SolderingMenu": {
      "text2": ["Soldering", "settings"],
      "desc": "Soldering settings"
    },
   ```

   instead of this:

   ```json
		"SolderingMenu": {
			"text2": [
				"Soldering",
				"settings"
			],
			"desc": "Soldering settings"
		},
   ```
2. I rearranged some of the *menu entries* in **`translation_EN.json`**, edited **`translations_def.js`** & **`Translation.h`** accordingly and wanted to use **`TranslationEditor`** to change all the other languages as well, so I would not have to do it by hand.

   **`translation_EN.json`** was used as ***Reference Language***.
And although the html view was matching the *new order*, the saved file and *view*, unfortunately maintained the *old order*!

Hopefully you are doing well and have time at hand to help me with that. 😊

thanks in advance
kind regards

**Comment by Ralim** [comment]
@discip  its would be better to put this into its own issue than bumping a really old thread.
I cant see any difference between the two you have posted other than formatting, so you might need to clarify what the difference is that you are looking for as well

**Comment by discip** [comment]
1. > other than formatting

   Yes, that's exactly it! Only the format.
The actual one takes up much space (scrolling wise). Some languages were formatted like that (narrower) before I updated them to match the output of **`TranslationEditor`**.
I did not want to open a separate issue, because I was not sure, if it would be liked / accepted.
2. Sorry, I don't know how to make a visual comparison between the two. 😅
   actual one:
https://github.com/Ralim/IronOS/blob/dev/Translations/translation_EN.json
**vs**
reordered one:
https://github.com/discip/IronOS/blob/test/Translations/translation_EN.json

**Comment by discip** [comment]
I did not want to open a separate issue, because I was not sure, if it would be liked / accepted.
If you still want this to be done, of course I will.

1. > other than formatting

   Yes, that's exactly it! Only the format.
The actual one takes up much space (scrolling wise). Some languages were formatted like that (narrower) before I updated them to match the output of **`TranslationEditor`**.

2. Sorry, I don't know how to make a visual comparison between the two. 😅
   actual one:
https://github.com/Ralim/IronOS/blob/dev/Translations/translation_EN.json
**vs**
reordered one:
https://github.com/discip/IronOS/blob/test/Translations/translation_EN.json

**Pull request merged**